nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -111,6 +116,9 @@ sub hasCPUFeature {
 push @kernelModules, "kvm-intel" if hasCPUFeature "vmx";
 push @kernelModules, "kvm-amd" if hasCPUFeature "svm";
 
+push @attrs, "hardware.cpu.amd.updateMicrocode = lib.mkDefault config.hardware.enableRedistributableFirmware;" if cpuManufacturer "AuthenticAMD";
+push @attrs, "hardware.cpu.intel.updateMicrocode = lib.mkDefault config.hardware.enableRedistributableFirmware;" if cpuManufacturer "GenuineIntel";
```

Comment:
Should we just include a module fo intel and kvm? Than we can add more stuff in the future without users having to regenerate it.

Reply:
so move https://github.com/NixOS/nixos-hardware/tree/master/common/cpu to https://github.com/NixOS/nixpkgs/tree/master/nixos/modules/hardware/cpu

Reply:
Agreed, right now we assume that all intel CPUs are desktop cpus but than there are xeons

Reply:
so should i create a intel cpu module? i dont think we will need one

Reply:
I guess we can go with this option than...